### ISSUE TITLE
GenericJMX can't autodetect the hostname (Patch available)

### ISSUE THREAD

**Issue by timurua** [collectd/collectd#182]
I am using GenericJMX to collect stats from java servers and transport them through write_graphite plugin. I noticed that if hostname isn't set in GenericJMX configuration, then the all the stats are reported under localhost hostname. This is not convinient especially if you have deployment of many servers. I made a patch so that GenericJMX can autodetect the hostname.

File: collectd-5.1.0/bindings/java/org/collectd/java/GenericJMXConfConnection.java
Diff (turn off highlighting):

```diff
214c214,219
<     pd.setHost ((this._host != null) ? this._host : "localhost");
---
>     String hostname = this._host;
>     try{
>         hostname = (hostname != null ? hostname : java.net.InetAddress.getLocalHost().getHostName());
>     } catch (Exception ex){}
>     hostname = (hostname != null ? hostname : "localhost");
>     pd.setHost (hostname);
```

**Comment by rubenk** [comment]
This seems fixed by f554ef29b3a20ff058878a3940e97d1dc21f1df0?

**Comment by octo** [comment]
ACK, seems fixed.
